nodebuilder/header/module.go

```diff
@@ -26,6 +26,15 @@ func ConstructModule(tp node.Type, cfg *Config) fx.Option {
 		fx.Supply(*cfg),
 		fx.Error(cfgErr),
 		fx.Supply(modp2p.BlockTime),
+		fx.Provide(
+			func(cfg *Config) []store.Option {
+				return []store.Option{
+					store.WithDefaultStoreCacheSize(cfg.Store.StoreCacheSize),
+					store.WithDefaultIndexCacheSize(cfg.Store.IndexCacheSize),
+					store.WithDefaultWriteBatchSize(cfg.Store.WriteBatchSize),
+				}
+			},
+		),
```

Comment:
One more thing. We really need to test that the provided `[]store.Option` is automatic and gets passed to the constructor's variadic `...store.Option` param. Right now, everything works, but it can false positive.

Reply:
@distractedm1nd has already created a task for unit testing.

Reply:
ok

Reply:
Also. On the line 26 the value of the config is provided, while the provide is expecting a pointer. This makes me wonder if the Provide for the options is called at all

Reply:
No. The validate func is called before we provide anything

Reply:
Double checked from the DASer PR, we are providing a slice, and I do remember reading in fx's documentation that variadic signatures are treated like slices, so we should be fine.

But nonetheless, @vgonkivs please test this. And if we have a unit test for it then that's gold!

Reply:
We now have it! And it turned out in our case that variadic signature is not treated like slices, we potentially missed something last time, @derrandz and should double check, and even better, write the same test like @vgonkivs did